src/main/java/com/ericsson/ei/subscription/SubscriptionRepeatDbHandler.java

```diff
@@ -82,70 +86,72 @@
      * Function that stores the matched aggregatedObjectId to the database.
      *
      */
-    public void addMatchedAggrObjToSubscriptionId(String subscriptionId, int requirementId, String aggrObjId)
-            throws Exception {
-
-        LOGGER.debug("Adding/Updating matched AggrObjId: " + aggrObjId + " to SubscriptionsId: " + subscriptionId
-                + " aggrId matched list");
+    public void addMatchedAggrObjToSubscriptionId(String subscriptionId, int requirementId, String aggrObjId) {
+        try {
```

Comment:
this try catch block is too large and contains code that may not throw this error. A better way is to extract the code that throws the error to it´s own method and add try catch in that.,

Reply:
I agree it is large, and I'm extracting parts into different, separate methods. I believe this work requires several iterations though and I'll try to limit the major changes to MongoDBHandler and SubscriptionRepeatDBHandler for this PR :)